### PR TITLE
Competition Result React improvements

### LINKED ISSUE #6018: MBLD old not displayed correctly in competition results table
See an example [here](https://www.worldcubeassociation.org/competitions/Euro2008/results/all?event=333mbo). It's been recently ported to React and this function is responsible for parsing MBLD attempt results:
 
https://github.com/thewca/worldcubeassociation.org/blob/ce57e2af345d7ad5cb4c0841315073b27eecaf5b/WcaOnRails/app/webpacker/javascript/wca-live/attempts.js#L20-L35

It should handle old-style format as well. The function was originally taken from wca-live, which doesn't have to support that format anyway.

### PR DESCRIPTION
**Pull request by Jambrose777**
- Added Record Type Labels to Single and Average Records
image

- Added Color to records (these colors match the Persons page Records
  - Notes: Ideally we should pull these colors from the same location. Due to this being in js and the other rails I wasnt too sure how to accomplish that.
image

- On Hover of a Country Flag, Display the Country Name
image

- Fixed #6018 MBO format
image

- Have Tied positions greyed out on subsequent occurances
image

- Fixed part of #5963 : When event nav overflows it will scroll
image

### CODE REVIEW

**Comment by Jambrose777** [comment]
I'd also like to include the () around the best and worst times. Ruby works like it adds a class best or worst to the individual times, and the parenthesis are added via the class, but I'm not sure how those classes are interpreted. We also have the times as a string instead of individual strings (so a class would be applied to all of them rn). 
Would it be best to try to replicate how the old system worked with classes, or should I add the parenthesis as part of the string that is made?

**Comment by viroulep** [comment]
The way it works in how results are formatted when displaying podiums for instance is that one `td` is generated per attempt, and the `best` or `worst` class is applied to them if appropriate.
You could reuse these css classes, but the thing here is that we just use one `td` for all attempts.

The easiest is probably to create a `formatAttemptsResult` that take a `result` parameter, and fix [this line](https://github.com/thewca/worldcubeassociation.org/blob/1f61f4689e09bd73fec8e6e2d70b80fa8550d140/WcaOnRails/app/webpacker/javascript/competition_results/index.js#L41) to use it.
I don't think we have something to sort attempts in our js codebase already (that you would need to determine the best/worst indexes); you can check how we compare `solve_times` [here](https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/lib/solve_time.rb#L146-L153).
My wild guess would be that you could do something like that:
```js
elements = result.attempts.filter((a) => a !== 0).map((a) => ({ value: a, dns: a === -2, dnf: a === -1 }));
_.sortBy(elements, ['skipped', 'dns', 'dnf', 'value']);
bestIndex = _.findIndex(attempts, elements[0].value);
worstIndex = _.findIndex(elements, elements[elements.length - 1].value);
```

And then iterate attempts with index (as you did for results), and add parenthesis if the index matches either of the best or worst.

By the way I think the tests are failing just because of the linter, you can probably fix most of them by running `bin/yarn run eslint -c .eslintrc.json app/webpacker/javascript/competition_results --fix`.
You can change the path to check/fix other files.

**Review by viroulep — CHANGES_REQUESTED** [review]
I finally had to some time to properly review this, thanks a lot for taking this on!
I left a few  comments on things I think deserves some changes, but the changes already look very good!

I also took a second thought at how we could format the best and worst attempt, and I actually think it would be more straightforward to just expose the best and worst index in the API to avoid duplicating the logic.
You can checkout this commit which implements it: https://github.com/viroulep/worldcubeassociation.org/commit/9dcd376470c2a9f01ea241815d2d8430c1d49934.
You can use it by downloading [the patch](https://github.com/viroulep/worldcubeassociation.org/commit/9dcd376470c2a9f01ea241815d2d8430c1d49934.patch) and running `git am 9dcd376470c2a9f01ea241815d2d8430c1d49934.patch` to apply it locally.

File: WcaOnRails/app/webpacker/javascript/event_navigation/index.scss [hunk 1]
```diff
@@ -9,4 +9,8 @@
 
 .selected {
     color: #fc4a0a;
-  }
\ No newline at end of file
+  }
+
+.event-menu-bar {
+  overflow: auto;
```
[the remark this answers is not in the document]
Comment (Jambrose777) [reply WcaOnRails/app/webpacker/javascript/event_navigation/index.scss:15]: I like this too! I will make the change.

**Comment by Jambrose777** [comment]
> I finally had to some time to properly review this, thanks a lot for taking this on!
> I left a few comments on things I think deserves some changes, but the changes already look very good!
> 
> I also took a second thought at how we could format the best and worst attempt, and I actually think it would be more straightforward to just expose the best and worst index in the API to avoid duplicating the logic.
> You can checkout this commit which implements it: [viroulep@9dcd376](https://github.com/viroulep/worldcubeassociation.org/commit/9dcd376470c2a9f01ea241815d2d8430c1d49934).
> You can use it by downloading [the patch](https://github.com/viroulep/worldcubeassociation.org/commit/9dcd376470c2a9f01ea241815d2d8430c1d49934.patch) and running `git am 9dcd376470c2a9f01ea241815d2d8430c1d49934.patch` to apply it locally.

Thanks Philippe! I've addressed everything and should be good to go now that I squashed the commits.

**Review by jonatanklosko — APPROVED** [review]
Great improvements, thanks @Jambrose777! Also thanks @viroulep for taking a look =)

**Pull request merged**